tests/unit/util/main.c

```diff
@@ -867,6 +867,28 @@ ZTEST(util, test_utf8_lcpy_truncated)
 	zassert_str_equal(dest_str, expected_result, "Failed to copy");
 }
 
+ZTEST(util, test_utf8_is_valid)
+{
+	/* Test whether the verification function meets the requirements */
+	zassert_true(utf8_is_valid("\xce\xba\xe1\xbd\xb9\xcf\x83\xce\xbc\xce\xb5", 11));
+	zassert_true(utf8_is_valid("\x00", 1));
+	zassert_true(utf8_is_valid("\xc2\x80", 2));
+	zassert_true(utf8_is_valid("\xef\xbf\xbf", 3));
+	zassert_true(utf8_is_valid("\xed\x9f\xbf", 3));
+	zassert_true(utf8_is_valid("\xed\xa0\x80", 3));
+	zassert_true(utf8_is_valid("\xed\xa0\x80\xed\xb0\x80", 6));
+	zassert_true(utf8_is_valid("\xef\xbf\xbe", 3));
+	zassert_false(utf8_is_valid("\x80", 1));
+	zassert_false(utf8_is_valid("\xc0", 1));
+	zassert_false(utf8_is_valid("\xfe", 1));
+	zassert_false(utf8_is_valid("\xff", 1));
+	zassert_false(utf8_is_valid("\xfe\xfe\xff\xff", 4));
+	zassert_false(utf8_is_valid("\xc0\x7f", 2));
+	zassert_false(utf8_is_valid("\xc0\xaf", 2));
+	zassert_false(utf8_is_valid("\xc1\xbf", 2));
+	zassert_false(utf8_is_valid("\xc0\x80", 2));
+}
```

Comment:
We should have tests where `maxlen` is larger and smaller than the provided string as well

Reply:
Yes, but you only need to add one. That is to test its behavior.

Reply:
I believe we should have all 3 cases:
1) maxlen smaller than strlen(str)
2) maxlen equal to strlen(str) (covered already)
3) maxlen larger than strlen(str)

Otherwise you aren't covering all checks in the code, right?

Reply:
If we use `strnlen`, I don't think we need to test when maxlen is greater than len.